src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/PropertyAccessorSymbolReferenceFinder.cs

```diff
@@ -1,47 +1,92 @@
 
-using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServices;
 using Microsoft.CodeAnalysis.Shared.Extensions;
-using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.FindSymbols.Finders
 {
     internal class PropertyAccessorSymbolReferenceFinder : AbstractMethodOrPropertyOrEventSymbolReferenceFinder<IMethodSymbol>
     {
         protected override bool CanFind(IMethodSymbol symbol)
-        {
-            return symbol.MethodKind.IsPropertyAccessor();
-        }
+            => symbol.MethodKind.IsPropertyAccessor();
 
         protected override async Task<ImmutableArray<SymbolAndProjectId>> DetermineCascadedSymbolsAsync(
             SymbolAndProjectId<IMethodSymbol> symbolAndProjectId,
             Solution solution,
             IImmutableSet<Project> projects,
+            FindReferencesSearchOptions options,
             CancellationToken cancellationToken)
         {
             var result = await base.DetermineCascadedSymbolsAsync(
-                symbolAndProjectId, solution, projects, cancellationToken).ConfigureAwait(false);
+                symbolAndProjectId, solution, projects, options, cancellationToken).ConfigureAwait(false);
 
-            var symbol = symbolAndProjectId.Symbol;
-            if (symbol.AssociatedSymbol != null)
+            // If we've been asked to search for specific accessors, then do not cascade.
+            // We don't want to produce results for the associated property.
+            if (!options.AssociatePropertyReferencesWithSpecificAccessor)
             {
-                result = result.Add(symbolAndProjectId.WithSymbol(symbol.AssociatedSymbol));
+                var symbol = symbolAndProjectId.Symbol;
+                if (symbol.AssociatedSymbol != null)
+                {
+                    result = result.Add(symbolAndProjectId.WithSymbol(symbol.AssociatedSymbol));
+                }
             }
 
             return result;
         }
 
-        protected override Task<ImmutableArray<Document>> DetermineDocumentsToSearchAsync(IMethodSymbol symbol, Project project, IImmutableSet<Document> documents, CancellationToken cancellationToken)
+        protected override async Task<ImmutableArray<Document>> DetermineDocumentsToSearchAsync(
+            IMethodSymbol symbol, Project project, IImmutableSet<Document> documents, 
+            FindReferencesSearchOptions options, CancellationToken cancellationToken)
         {
-            return FindDocumentsAsync(project, documents, cancellationToken, symbol.Name);
+            var result = await FindDocumentsAsync(
+                project, documents, cancellationToken, symbol.Name).ConfigureAwait(false);
+
+            if (options.AssociatePropertyReferencesWithSpecificAccessor)
```

Comment:
>AssociatePropertyReferencesWithSpecificAccessor [](start = 24, length = 47)

I need to think more about this. My intuition was to expect fewer documents (not more), when the option to do a narrower search is set.

Reply:
so, this is specifically teh codepath that finds "property **accessors**".  The existing code actually only exists to find things like explicit calls to ```get_Foo(1, 2, 3)```.  i.e. when C# accesses a VB indexed property.   

--

So this is purely additive here.  We still find those as before, but we also find references to accessors in things like ```this.Foo++```.